lib/src/configcat_client.dart

```diff
@@ -260,18 +291,39 @@ class ConfigCatClient {
 
       for (final entry in result.settings.entries) {
         if (entry.value.variationId == variationId) {
-          return MapEntry(entry.key, entry.value.value);
+          return MapEntry(
+              entry.key,
+              _parseSettingValue<T>(
+                  entry.value.settingsValue, entry.value.type));
         }
 
-        for (final rolloutRule in entry.value.rolloutRules) {
-          if (rolloutRule.variationId == variationId) {
-            return MapEntry(entry.key, rolloutRule.value);
+        for (final targetingRule in entry.value.targetingRules) {
+          if (targetingRule.servedValue != null &&
+              targetingRule.servedValue?.variationId == variationId) {
+            return MapEntry(
+                entry.key,
+                _parseSettingValue<T>(targetingRule.servedValue!.settingsValue,
+                    entry.value.type));
+          }
+          var targetRulePercentageOptions = targetingRule.percentageOptions;
+          if (targetRulePercentageOptions != null) {
+            for (final percentageOption in targetRulePercentageOptions) {
+              if (percentageOption.variationId == variationId) {
+                return MapEntry(
+                    entry.key,
+                    _parseSettingValue<T>(
+                        percentageOption.settingsValue, entry.value.type));
+              }
+            }
```

Comment:
Once we've taken the served value path, we don't want to end up on the % options one if the variation id is missing. (Unfortunately, this issue is also present in the Java and Android SDK. Please correct those as well.)

```suggestion
          if (targetingRule.servedValue != null) {
            if (targetingRule.servedValue?.variationId == variationId) {
              return MapEntry(
                entry.key,
                _parseSettingValue<T>(targetingRule.servedValue!.settingsValue,
                    entry.value.type));
            }
          } else {
            var targetRulePercentageOptions = targetingRule.percentageOptions;
            if (targetRulePercentageOptions != null) {
              for (final percentageOption in targetRulePercentageOptions) {
                if (percentageOption.variationId == variationId) {
                  return MapEntry(
                      entry.key,
                      _parseSettingValue<T>(
                          percentageOption.settingsValue, entry.value.type));
                }
              }
            }
```